addons/web/static/src/js/views/calendar/calendar_model.js

```diff
@@ -404,7 +405,9 @@ return AbstractModel.extend({
             monthNamesShort: moment.monthsShort(),
             dayNames: moment.weekdays(),
             dayNamesShort: moment.weekdaysShort(),
-            firstDay: _t.database.parameters.week_start,
+            // Index for Sunday is 0 in calendar, but week_start index for
+            // Sunday is 7, so we map it to calendar indexing here.
+            firstDay: week_start != 7 ? week_start : 0,
```

Comment:
You could have everything on one line:
`firstDay: _t.database.parameters.week_start % 7,  // calendar uses index 0 for Sunday but Odoo stores it as 7`